refl1d/webview/client/src/main.js

```diff
@@ -1,12 +1,21 @@
 import "bootstrap/dist/css/bootstrap.min.css";
 import { computed, createApp } from "vue";
-import { file_menu_items, fileBrowser, shared_state, socket } from "bumps-webview-client/src/app_state";
+import { file_menu_items, fileBrowser, shared_state } from "bumps-webview-client/src/app_state";
 import App from "bumps-webview-client/src/App.vue";
+import { io } from "socket.io-client";
 import { dqIsFWHM } from "./app_state";
 import { panels } from "./panels";
 import "./style.css";
 
 const name = "Refl1D";
+const urlParams = new URLSearchParams(window.location.search);
+const singlePanel = urlParams.get("single_panel");
+const sio_base_path = urlParams.get("base_path") ?? window.location.pathname;
+const sio_server = urlParams.get("server") ?? "";
+
+const socket = io(sio_server, {
+  path: `${sio_base_path}socket.io`,
+});
```

Comment:
This duplicates code in bumps client main.js. Can it be moved into a bumps library function callable from refl1d?

Reply:
not easily...